abci/abci.go

```diff
@@ -157,6 +187,20 @@ func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 		for ; iterator != nil; iterator = iterator.Next() {
 			memTx := iterator.Tx()
 
+			// Gas Limit check
+			feeTx, ok := memTx.(sdk.FeeTx)
+			if !ok {
+				txsToRemove[memTx] = struct{}{}
+				continue selectTxLoop
+			}
+
+			gasLimit := feeTx.GetGas()
+			if updatedGasLimit := int64(gasLimit) + totalGasLimit; updatedGasLimit > maxGasLimit {
+				continue selectTxLoop
+			} else {
+				totalGasLimit = updatedGasLimit
```

Comment:
Shld we only update this after we know we can insert the tx? Like what if the tx was alr inserted into the proposal, or it fails to decode, etc.) those conditions are checked after this? But the gas total is retained?